reactjs/react2/week1/new-app/app/nasaepic/epic/page.js

```diff
@@ -0,0 +1,52 @@
+
+"use client"
+import { useRouter } from "next/navigation";
+import { useEffect, useState } from "react";
+import Navbar from "@/components/Navbar";
+
+const NASA_EPIC_API_KEY = process.env.NEXT_PUBLIC_API_KEY;
+
+const EpicImage = () => {
+  const router = useRouter();
+  const [date, setDate] = useState("");
+  const [imageUrl, setImageUrl] = useState("");
+
+  useEffect(() => {
+    if (router.isReady) {
+      const { date } = router.query;
+      setDate(date);
+
+      if (date) {
+        fetch(
+          `https://api.nasa.gov/EPIC/api/natural/date/${date}?api_key=${NASA_EPIC_API_KEY}`
+        )
+          .then((response) => response.json())
+          .then((data) => {
+            if (data && data.length > 0) {
+              const image = data[0];
+              const dateParts = image.date.split(" ")[0].split("-");
+              const formattedDate = `${dateParts[0]}/${dateParts[1]}/${dateParts[2]}`;
+              const url = `https://epic.gsfc.nasa.gov/archive/natural/${formattedDate}/png/${image.image}.png`;
+              setImageUrl(url);
+            }
+          })
+          .catch((err) => console.error(err));
+      }
+    }
+  }, [router.isReady, router.query]);
+
+ 
+
+  return (
+    <div>
+      <Navbar />
```

Comment:
Good job on adding Navbar! It looks very nice.

Nevertheless, puting repeatable components to every `page.js` is considered not a good practice: the component will be new every time new page is loading and it'll end up doing "hide and appear" all the time, which might be annoying

For reusing common parts of the application on every page, it is better to put them in `layout.js` file: this way it won't be updated or hidden every time user goes to another page